.builders/upload.py

```diff
@@ -114,21 +129,21 @@ def main():
             else:
                 # https://packaging.python.org/en/latest/specifications/binary-distribution-format/#file-name-convention
                 name, version, python_tag, abi_tag, platform_tag = wheel.stem.split('-')
-                build_number = 0
-                while True:
-                    artifact_name = f'{name}-{version}-{build_number}-{python_tag}-{abi_tag}-{platform_tag}.whl'
-                    artifact = bucket.blob(f'{artifact_type}/{project_name}/{artifact_name}')
-                    if not artifact.exists():
-                        break
-
-                    build_number += 1
+                existing_wheels = list(bucket.list_blobs(prefix=f'{artifact_type}/{project_name}/{name}-{version}-'))
```

Comment:
We can't guarantee the name nor the version is normalized in the wheel file name so this wouldn't work

Reply:
e.g. https://pypi.org/project/PyYAML/#files

Reply:
I'm not sure we'd be making assumptions about name normalization here. The `{name}` here should be the same every time we upload the same package, therefore we should be able to retrieve the names for matching wheels this way.

As far as I can tell lines 119-122 of the old file make the same exact assumption.